From 2eb4da7b78aa7ed9a7556552f08aa6679e566b52 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 23 Sep 2017 10:59:07 +0200
Subject: [PATCH 1/1] efi_loader: incorrect definition of
 EFI_SIMPLE_NETWORK_PROTOCOL

WaitForPacket is an event and not a function pointer.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/efi_api.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/efi_api.h b/include/efi_api.h
index a83d94621e..0978d9ace7 100644
--- a/include/efi_api.h
+++ b/include/efi_api.h
@@ -644,7 +644,7 @@ struct efi_simple_network
 			ulong *header_size, ulong *buffer_size, void *buffer,
 			struct efi_mac_address *src_addr,
 			struct efi_mac_address *dest_addr, u16 *protocol);
-	void (EFIAPI *waitforpacket)(void);
+	struct event *wait_for_packet;
 	struct efi_simple_network_mode *mode;
 };
 
-- 
2.11.0

